crates/workspace/src/pane.rs

```diff
@@ -2544,7 +2544,9 @@ impl Pane {
                             .read(cx)
                             .item_for_entry(entry, cx)
                             .and_then(|item| item.project_path(cx))
-                            .map(|project_path| project_path.path);
+                            .map(|project_path| project_path.path)
+                            .filter(|path| !path.as_os_str().is_empty());
+                        let has_relative_path = relative_path.is_some();
```

Comment:
```suggestion
                        let visible_in_project_panel = relative_path.is_some()
                            && worktree.is_some_and(|worktree| worktree.read(cx).is_visible());
```

We could do
```rs
let project_path = pane
    .read(cx)
    .item_for_entry(entry, cx)
    .and_then(|item| item.project_path(cx));
let worktree = project_path.as_ref().and_then(|project_path| {
    pane.read(cx)
        .project
        .upgrade()?
        .read(cx)
        .worktree_for_id(project_path.worktree_id, cx)
});
```

a bit above this and check that we're displaying anything in the panel to reveal.